src/minetest.cpp

```diff
@@ -166,6 +163,31 @@ bool Minetest::runMod(EditorState *state, const std::string &world)
 	FileFormat *writer = getFromType(FILE_FORMAT_LUA, state);
 	save_file(writer, state, mod_to + "init.lua");
 
+	// Build argument list
+	char *argv[9];
+	argv[0] = (char*)(minetest_exe).c_str();
+	argv[1] = "--worldname";
+	argv[2] = (char*)world.c_str();
+	argv[3] = "--name";
+	argv[4] = "tester";
+	argv[5] = "--address";
+	argv[6] = "";
+	argv[7] = "--go";
+	argv[8] = NULL;
+
+#ifndef _WIN32
+	pid_t pid = fork();
+	if (pid == 0) {
+		chdir(minetest_dir.c_str());
+		std::cerr << "FORK: launching mt, exe = " << argv[0] << std::endl;
+		if (execvp(argv[0], argv)) {
+			std::cerr << "Couldn't run minetest!" << std::endl;
+		}
+		exit(0);
```

Comment:
Should use `_exit(1)` instead. Yes, the underscore is intentional. See [this answer](http://stackoverflow.com/a/2329754) for explanation.